security/src/main/scala/pl/touk/nussknacker/ui/security/api/ImpersonationSupport.scala

```
@@ -0,0 +1,11 @@
+package pl.touk.nussknacker.ui.security.api
+
+trait ImpersonationSupport {
+  def getImpersonatedUserData(impersonatedUserIdentity: String): Option[ImpersonatedUserData]
+}
+
+trait NoImpersonationSupport extends ImpersonationSupport {
+  override def getImpersonatedUserData(impersonatedUserIdentity: String): Option[ImpersonatedUserData] = None
```

I think that we should rather return `Left[ImpersonationNotSupported]` or at least throw an exception in this case instead of retuning `None`. `None` suggests that it is a client problem, that sent id of user which doesn't exists but it might be either client problem that uses impersonation but it shouldn't or server problem, that configuration is invalid.

---

I intorduced a solution based on `Either` instead of exceptions.
We didn't want to change `getImpersonatedUserData` signature to `Either[NoImpersonationSupport, Option[ImpersonatedUserData]]` as while implementing `getImpersonatedUserData` we should not care about the `NoImpersonationSupport` situation. So I introduced `getImpersonatedUserDataWithSupportCheck` implemented for both `ImpersonationSupport` and `NoImpersonationSupport` which wraps `getImpersonatedUserData` helping with error handling.

I also added custom responses for both Tapir and Akka in case somebody tries to impersonate while authentication mechanism does not support it.